redbot/core/commands/converter.py

```diff
@@ -225,17 +225,28 @@ class TimedeltaConverter(dpy_commands.Converter):
         If provided, you can constrain a user to expressing the amount of time
         in specific units. The units you can chose to provide are the same as the
         parser understands. `weeks` `days` `hours` `minutes` `seconds`
+    default_unit : Optional[str]
+        If provided, it will additionally try to match integer-only input into
+        a timedelta, using the unit specified. Same units as in `allowed_units`
+        apply.
     """
 
-    def __init__(self, *, minimum=None, maximum=None, allowed_units=None):
+    def __init__(self, *, minimum=None, maximum=None, allowed_units=None, default_unit=None):
         self.allowed_units = allowed_units
+        self.default_unit = default_unit
         self.minimum = minimum
         self.maximum = maximum
 
     async def convert(self, ctx: "Context", argument: str) -> timedelta:
-        delta = parse_timedelta(
-            argument, minimum=self.minimum, maximum=self.maximum, allowed_units=self.allowed_units
-        )
+        if self.default_unit and argument.isdigit():
```

Comment:
```suggestion
        if self.default_unit and argument.isdecimal():
```

`isdigit` isn't safe for this.

for why, consider `³` . (In case of font display issues, the equivalent escaped string is `\N{SUPERSCRIPT THREE}`)

This passes both `isnumeric` and `isdigit` but will raise a ValueError if you attempt to make it an int.